File: MgDev/UnitTest/WebTier/DotNet/TestCommon/PlatformApiTestExecutor.cs
using SqliteDotNet;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OSGeo.MapGuide.Test.Common
{
    /// <summary>
    /// The base class of all MapGuide API test executors
    /// </summary>
    public abstract class PlatformApiTestExecutor : TestExecutorBase
    {
        protected string _unitTestVmPath;
        private SqliteDb _unitTestDb;
        protected SqliteVm _unitTestVm;

        protected PlatformApiTestExecutor(string opName, string apiType, string unitTestVm)
        {
            _opName = opName;
            _apiType = apiType;
            _unitTestVmPath = unitTestVm;

            _unitTestDb = new SqliteDb();
            _unitTestDb.Open(_unitTestVmPath);

            _unitTestVm = new SqliteVm(_unitTestDb, true);
        }

        public override void Dispose()
        {
            _unitTestVm.SqlFinalize();
            _unitTestVm = null;
            try
            {
                _unitTestDb.Close();
            }
            catch { }
            _unitTestDb = null;
        }

        private string _apiType;

        public override string Api
        {
            get { return _apiType; }
        }

        private string _opName;

        public override string OperationName
        {
            get { return _opName; }
        }
    }
}

File: MgDev/UnitTest/WebTier/Java/src/org/osgeo/mapguide/test/common/PlatformApiTestExecutor.java
package org.osgeo.mapguide.test.common;

import java.io.File;
import java.util.HashMap;
import com.almworks.sqlite4java.*;

public abstract class PlatformApiTestExecutor extends TestExecutorBase
{
    private String _opName;
    private String _apiType;
    protected SQLiteConnection _conn;

    protected PlatformApiTestExecutor(String opName, String apiType, String dbPath)
    {
        _opName = opName;
        _apiType = apiType;
        try {
            _conn = new SQLiteConnection(new File(dbPath));
            _conn.open(false);
        } catch (Exception ex) {
            throw new RuntimeException(ex);
        }
    }

    protected void ReadParameterValue(int paramSetId, String paramName, HashMap<String, String> paramValues) {
        CommonUtility.ReadParameterValue(_conn, paramSetId, paramName, paramValues);
    }

    protected void ReadParameterValue(int paramSetId, String paramName, HashMap<String, String> paramValues, boolean isPath) {
        CommonUtility.ReadParameterValue(_conn, paramSetId, paramName, paramValues, isPath);
    }

    public void close()
    {
        if (_conn != null) {
            _conn.dispose();
            _conn = null;
        }
    }

    public String getApi()
    {
        return _apiType;
    }

    public String getOperationName()
    {
        return _opName;
    }
}